src/components/SignIn/useSignIn.ts

```diff
@@ -0,0 +1,38 @@
+import axios from "../../utils/axios";
+import { useState } from "react";
+import { useRouter } from "next/router";
+import { useToast } from "@chakra-ui/react";
+
+export default function useSignIn() {
```

Comment:
Is there any specific reason for you to extract a custom hook here? Since it is not reusable, you must have a good reason to do so.

Reply:
to manage all the API

Reply:
It is true that building customHooks can let you extract component logic into reusable functions. you can also separate logic and UI to improve the reading experience (of course, this varies from person to person, some people think that if this logic is only used in this component, then Just put it in there),
In addition, another reason to separate the logic is to use renderHook (from @testing-library/react) to test this logic separately during testing